pkg/executer/executer.go

```diff
@@ -44,7 +45,17 @@ func (e *CommonExecuter) execute(cmd *exec.Cmd) (stdout string, stderr string, e
 	var stdoutBytes, stderrBytes bytes.Buffer
 	cmd.Stdout = &stdoutBytes
 	cmd.Stderr = &stderrBytes
+	// Set Pdeathsig to SIGTERM to kill the process and its children when the parent process is killed.
+	// This should prevent orphaned processes and allow for the sub process to gracefully terminate.
+	// ref. https://github.com/golang/go/blob/release-branch.go1.21/src/syscall/exec_linux.go#L91
+	cmd.SysProcAttr = &syscall.SysProcAttr{Pdeathsig: syscall.SIGTERM}
```

Comment:
can we add the same thing to console:

https://github.com/flightctl/flightctl/blob/main/internal/agent/device/console.go#L204

So when we are killed and the console is running we don't leave any orphans, etc.